deps/exokit-bindings/webglcontext/src/webgl.cc

```diff
@@ -4938,12 +4997,33 @@ NAN_METHOD(WebGLRenderingContext::GetSupportedExtensions) {
   // glGetIntegerv(GL_NUM_EXTENSIONS, &numExtensions);
 
   int numExtensions = sizeof(webglExtensions)/sizeof(webglExtensions[0]);
-  Local<Array> result = Nan::New<Array>(numExtensions);
+  Local<Array> result = Nan::New<Array>(0);
+
+  GLint n=0;
+  glGetIntegerv(GL_NUM_EXTENSIONS, &n);
+
+  for (GLint i=0; i<n; i++)
+  {
+    const char* extension = (const char*)glGetStringi(GL_EXTENSIONS, i);
+
+    for (GLint j = 0; j < numExtensions; j++) {
+      bool matchesSupportedExtension = (strcmp(extension+3,webglExtensions[j]) == 0);
```

Comment:
What is `+3`? Is that guaranteed to never overflow?

Reply:
`+3` is to compare against `weblExtensions` (which do not lead with `GL_`) against the GL_EXTENSIONS (which have `GL_` as the first 3 characters). See example output here: https://www.geeks3d.com/20100722/tips-how-to-get-the-list-of-the-opengl-extensions-with-a-core-profile/

![gloutputCapture](https://user-images.githubusercontent.com/29695350/58847496-cd33c480-8648-11e9-8446-e6949032a8ef.PNG)

Reply:
There aren't any obvious cases where it would overflow, based on the values it outputs. (`GL_xxx`)

Reply:
It looks like there is also `WGL_`, so a regex that filters for characters up to first `_` might be better.

Reply:
The problem is that there is no guarantee this code won't explode if some driver decides a different format for an extension string. Can this code be written without making any assumptions about the GL implementation?

Reply:
The first thing that comes to mind is to see if characters of `webglExtension[i]` are found in `extension`, but that would be assuming things as well.

Reply:
What about if we used `std::string` and did a safe substring? That probably wouldn't crash regardless of what the driver returns.

Reply:
The list from https://www.geeks3d.com/20100722/tips-how-to-get-the-list-of-the-opengl-extensions-with-a-core-profile/  has multiple substrings, would cases like that be relevant here?:
```
GL extension 191: GL_NV_texture_shader
GL extension 192: GL_NV_texture_shader2
GL extension 193: GL_NV_texture_shader3
GL extension 194: GL_NV_transform_feedback
GL extension 195: GL_NV_transform_feedback2
GL extension 196: GL_NV_vertex_array_range
GL extension 197: GL_NV_vertex_array_range2
GL extension 200: GL_NV_vertex_program
GL extension 201: GL_NV_vertex_program1_1
GL extension 202: GL_NV_vertex_program2
GL extension 203: GL_NV_vertex_program2_option
GL extension 204: GL_NV_vertex_program3
```

Reply:
Yes, quite possibly.

Though I would expect the webgl extensions to be listed by the driver under a non-prefixed form, which would be the one we need to check.

Reply:
note from discord:
>  Can probably just hardcode on compile for multiview extension support in the list / hardcode as available in all cases, since that seems to be the case

Reply:
Based on the note^ 436dba4